tests/framework/umbra_app.c

```diff
@@ -0,0 +1,167 @@
+/* **********************************************************
+ * Copyright (c) 2017 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#ifndef ASM_CODE_ONLY /* C code ***********************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+void asm_test(void);
+#ifdef ARM
+void asm_test_thumb(void);
+#endif
+
+int
+main(int argc, char *argv[])
+{
+    asm_test();
+#ifdef ARM
+    asm_test_thumb();
+#endif
+    printf("TEST PASSED\n");
+    return 0;
+}
+
+#else /* asm code *************************************************************/
+#include "cpp2asm_defines.h"
+#include "umbra_test_shared.h"
+START_FILE
+
+#ifdef X64
+# define FRAME_PADDING 8
+#else
+# define FRAME_PADDING 0
+#endif
+
+#ifdef ARM
+.code 32 /* ARM */
+#endif
+
+#define FUNCNAME asm_test
+        DECLARE_FUNC(FUNCNAME)
+GLOBAL_LABEL(FUNCNAME:)
+#ifdef X86
+        /* push callee-saved registers */
+        PUSH_SEH(REG_XBX)
+        PUSH_SEH(REG_XBP)
+        PUSH_SEH(REG_XSI)
+        PUSH_SEH(REG_XDI)
+        sub      REG_XSP, FRAME_PADDING /* align */
+        mov 	 REG_XBP, REG_XSP
+        sub		 REG_XSP, 4
```

Comment:
The operand alignment looks off here and on the line above.  The style guide says "place opcodes on column 8 and start operands on column 17" -- hard to tell from this view which one of lines 78 and 79 is correct.

Reply:
Done. Looks like this was a tabs vs spaces confusion issue.